### PR TITLE
ENH: UpdateComponent, lock down mirror offsets and others

### PR DESCRIPTION
**Pull request by ZLLentz**
## Description

- Add `UpdateComponent`, a `Component` subclass that allows you to inherit and update a super class's components.
- Add `BeckhoffAxisNoOffset`, a variation on `BeckhoffAxis` that uses `UpdateComponent` to set the user offset related signals to read-only.
- Apply `BeckhoffAxisNoOffset` to all static beamline motors.

## Motivation and Context

For the beamline Beckhoff axes, we don't want to be changing their directions or offsets accidentally. In fact these should always always always be 0/POS in 99% of cases, and some other static value in the other 1% of cases. An easy/clean way to handle this is to prevent the Python from writing to them by making them `EpicsSignalRO`. This has to be specifically targeted at specific instances, both for backcompat and for the end stations that do want to make use of the EPICS-level direction and offset controls.

Matt Seaberg originally pointed this out for the mirrors in https://jira.slac.stanford.edu/browse/LCLSPC-131, I have extended this to all beamline axes that I deemed appropriate.

To implement this, I decided to finally create the thing I've wanted for a while: an easy way to inherit/edit components inside a device definition, the `UpdateComponent`. I think this implementation minimizes the duplication of typing and feels natural to use.

## How Has This Been Tested?

Only interactively, so it is possible that I have broken something- hopefully the tests will catch it.
Before merging I need to add some tests for `UpdateComponent` (which I plan to upstream to `ophyd`).

## Where Has This Been Documented?

Not enough yet

## Pre-merge checklist
- [x] Code works interactively
- [x] Code contains descriptive docstrings, including context and API
- [x] New/changed functions and methods are covered in the test suite where possible
- [x] Test suite passes locally
- [x] Test suite passes on travis
- [x] Ran docs/pre-release-notes.sh and created a pre-release documentation page
- [x] Pre-release docs include context, functional descriptions, and contributors as appropriate

### CODE REVIEW

**Comment by ZLLentz** [comment]
I won't merge this until next week when I'll have finished the tests and Ken has had a chance to review after his vacation, but I might put this into dev pcdsdevices to get the typhos screen update out if the test suite passes.

File: pcdsdevices/device.py
Comment (klauer) [inline pcdsdevices/device.py:303]: I think this is good enough for the majority of cases, but you might search the `mro` to deal with multiple inheritance.

**Review by klauer — COMMENTED** [review]
This is pretty neat! I think it's implemented cleanly and just had one minor suggestion.
I'm assuming that since this is already in use, it's working well.

Aside from the necessary documentation steps, I can't see anything else that might need to be done here. Will wait for that to approve.

File: pcdsdevices/device.py
Comment (ZLLentz) [reply pcdsdevices/device.py:303]: Ahh, good point, multiple inheritance breaks the simple grab here.

Comment (ZLLentz) [reply pcdsdevices/device.py]: Fixed this now by iterating through `mro()[1:]` (skip own class)

**Review by klauer — APPROVED** [review]
Nothing to add from previous review, as you resolved the one thing I noticed. Looks good 👍

**Pull request merged**
